### ISSUE TITLE
Improving Prophet usage even further 

### ISSUE THREAD

**Issue by ZED711** [facebook/prophet#629]
Hello everyone. 
I'm currently doing a project with several sets of data input. Some of them I do have a decent error, while others are not that good. I **have added some regressors with the Fourier series** (period = 7 and series order = 10, which seems to be a good bet according to this paper in page 11/12: https://peerj.com/preprints/3190/). Plus, I **even calculate the outliers of the data and remove them**, as it is recommended in the Prophet platform (https://facebook.github.io/prophet/docs/outliers.html). For each set of data, I tried as well several values for the variable **'changepoint_prior_scale'** to get the best plot. All of the above helped to reduce the error. 
This is my best result: 

![best set](https://user-images.githubusercontent.com/41108785/43452524-91853bfc-94af-11e8-82af-ba30c7d527d2.PNG)
I used as input 3 weeks of data (I have 4) and forecasted the 4th week. Then, I compared the results and calculated the error between samples. The median error was 12% and the average error was 20% (there are some samples with a huge error that raise the average error (that's why I calculated the median). Plus, the plot seems legit. 
However, there are other sets of data that do have a higher error. In the following case, I've got a plot with 166% of average error and a median error of 40.5%. 

![bad](https://user-images.githubusercontent.com/41108785/43452409-47c6fa78-94af-11e8-9ac0-d1bce2fda064.PNG)

**I tried to use manual change points and mess around with changepoint_range but didn't get any impact at all.** 
Is there anything else that I can do to improve this project even further while using the Prophet? Maybe, there are some data sets that don't have enough samples or there isn't a pattern in those samples for Prophet to do a good forecast, that might be the case, right?

**Comment by vhpietil** [comment]
Unfortunately I do not have a way to make your forecasts with those problematic datasets better. However, I would like to point out one (maybe obvious) thing. The second example contains values that are way lower than in the first example. In that case, even if absolute errors are similar, percentage errors are much larger. If observation is very close to zero, even slightest error in the prediction gives huge percentage error. That does not neccessary mean that the prediction is bad. 

I do not know your application and if this is any helpful. Maybe percentage error is all that matters to you and if that is the case then you need to do everything you can to reduce that error.  But I have noticed in my own work that often largest percentage errors are due to small actual values and the predictions might actually be pretty good. I need to use different metrics depending on the scale of the actual values I am trying to predict. 

You say about the first plot that it seem legit. In my eyes the second plot seems legit too. But I don't know your application. Are you able to tell us about your field? What is this data and why are you predicting it? 

tl;dr Maybe percentage error is not a good metric for all data sets, maybe it is :)

**Comment by ZED711** [comment]
I'm predicting volume data (in bits) of several base stations in the United Kingdom. The objective is to make 1 month of forecast with 1 month of data to help the company that I'm working with to know in advance the amount of users that are using this base stations. I'm still trying to improve my code and reduce the error as much as possible before doing the final forecast.

**Comment by vhpietil** [comment]
I sounds like the absolute error is important to you as well. But I understand the percentage errors might be easier to communicate.

**Comment by ZED711** [comment]
Exactly, it's kind of a guarantee! Do you have any idea of other things that I could try to implement to improve the forecast? I've read the Prophet documentation and I think that I can't do much better than this... (implemented the Fourier Series, removed outliers, tried different values for changepoint_priorscale)

**Comment by bletham** [comment]
The bottom time series has those really high values (> 6) that are not being fit by the model. The reason they're not being captured is because they don't happen every day (so it's not part of the daily seasonality). Is there something special about those days that you know? Some type of known holidays or periodicity? Some way to have a feature that would allow the model to determine if a day should have a high value or not?
